src/hotspot/os/windows/osThread_windows.hpp

```diff
@@ -27,7 +27,7 @@
 
   typedef void* HANDLE;
  public:
-  typedef unsigned long thread_id_t;
+  typedef unsigned int thread_id_t;
```

Comment:
Since Windows is LLP64, this has no effect, even though beginthreadex returns an unsigned int.

Reply:
It's also another formatting change, as this is printed as %d in shared code. This also matches the definition of int on other platforms. This should functionally be harmless, because ints are promoted to longs (DWORD) whenever thread_id_t is passed to a callee that expects a DWORD, and since longs are the same size as ints this is effectively an implicit cast with no effect in those cases

Reply:
I can live with this change. Its also small and isolated.